tests/gtests/sycl/api/CMakeLists.txt

```diff
@@ -28,7 +28,7 @@ if(DNNL_WITH_SYCL)
     endif()
 
     # Enable linking SYCL kernels.
-    if(DNNL_SYCL_CUDA OR (DNNL_SYCL_GENERIC AND NVIDIA_TARGET_SUPPORTED))
+    if(DNNL_SYCL_GENERIC AND NVIDIA_TARGET_SUPPORTED)
```

Comment:
Could you explain, please, why `DNNL_SYCL_CUDA` is removed?

Reply:
The tests `Simple/sycl_memory_buffer_test.InteropReorderAndUserKernel/cpu` and `Simple/sycl_memory_buffer_test.EltwiseWithUserKernel/cpu` run with a CPU engine and try to run a SYCL kernel on CPU. They currently fail because the nvidia target triple is set. This results in a `UR_ERROR_INVALID_BINARY`. I am not sure if this is intended behaviour or the tests should be skipped, but removing the `DNNL_SYCL_CUDA` from the check here allows them to run successfully on CPU as it results in not setting the nvidia target triple.

Reply:
What you are saying is that target shouldn't be set for CPU part of the test. You removed linker options for pure Nvidia target, but how this change would help for a generic case targeting Nvidia backend?

Reply:
@sgeor255, I think the problem arises because due to overriding the default SYCL targets with `-fsycl-targets=nvptx64-nvidia-cuda` which throws off the default SYCL target, which is `spir64`. We could try to use both targets here as we expect the kernels to be compiled for CPU as well:
`-fsycl-targets=nvptx64-nvidia-cuda,spir64`